lib/service/service.go

```diff
@@ -355,6 +356,13 @@ func (process *TeleportProcess) addConnector(connector *Connector) {
 	process.connectors[connector.ClientIdentity.ID.Role] = connector
 }
 
+func (process *TeleportProcess) setAuthServerFeatures(features proto.Features) {
+	process.Lock()
+	defer process.Unlock()
+
+	process.Config.AuthServerFeatures = features
```

Comment:
Hmm, I'm not sure why this sets it in the `process.Config`? I would probably make an internal field TeleportProcess instead and initialize that (similar to `process.connectors` above).

Reply:
revised